Source/Scene/Expression.js

```diff
@@ -773,7 +774,9 @@ define([
         var evaluate = unaryFunctions[call];
         return function(feature) {
             var left = this._left.evaluate(feature);
-            if (typeof left === 'number') {
+            if (call === 'noise' && left instanceof Cartesian3) {
+                return Cartesian3.fromElements(evaluate(left.x), evaluate(left.y), evaluate(left.z), ScratchStorage.getCartesian3());
+            } else if (typeof left === 'number') {
```

Comment:
The validation check would be better below since noise with the wrong argument type deserves its own error message.